src/SharpGLTF.Core/Schema2/agi.Articulations.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Linq;
+using SharpGLTF.Collections;
+
+namespace SharpGLTF.Schema2
+{
+    public sealed partial class AgiRootArticulations
+    {
+        internal AgiRootArticulations(ModelRoot root)
+        {
+            _articulations = new ChildrenCollection<AgiArticulation, AgiRootArticulations>(this);
+        }
+
+        protected override IEnumerable<ExtraProperties> GetLogicalChildren()
+        {
+            return base.GetLogicalChildren().Concat(_articulations);
+        }
+
+        public IReadOnlyList<AgiArticulation> Articulations => _articulations;
+
+        public AgiArticulation CreateArticulation(string name)
+        {
+            var articulation = new AgiArticulation();
+            articulation.Name = name;
+
+            _articulations.Add(articulation);
+
+            return articulation;
+        }
+    }
+
+    public sealed partial class AgiNodeArticulations
+    {
+        internal AgiNodeArticulations(Node node) { }
+
+        public String ArticulationName
+        {
+            get => _articulationName;
+            set => _articulationName = value;
+        }
+
+        public Boolean? IsAttachPoint
+        {
+            get => _isAttachPoint;
+            set => _isAttachPoint = value;
+        }
+    }
+
+    public sealed partial class AgiArticulation : IChildOf<AgiRootArticulations>
+    {
+        internal AgiArticulation()
+        {
+            _stages = new ChildrenCollection<AgiArticulationStage, AgiArticulation>(this);
+        }
+
+        protected override IEnumerable<ExtraProperties> GetLogicalChildren()
+        {
+            return base.GetLogicalChildren().Concat(_stages);
+        }
+
+        public IReadOnlyList<AgiArticulationStage> Stages => _stages;
+
+        public AgiArticulationStage CreateArticulationStage(string name)
+        {
+            var stage = new AgiArticulationStage();
+            stage.Name = name;
+
+            _stages.Add(stage);
+
+            return stage;
+        }
+
+        public String Name
+        {
+            get => _name;
+            set => _name = value;
+        }
+
+        public Vector3? PointingVector
```

Comment:
Same nullable issue commented above. Sometimes the schema code generator already defines the default value, I guess in this case would be (0,0,0) ?? which would be an invalid pointing vector anyway.

Reply:
This one is a special case, where `null` itself is the default value, and that value has a specific meaning of turning off the feature.

I've added some guards around the setter here, to keep things a little more consistent.